test/app/plugins.js

```diff
@@ -0,0 +1,21 @@
+require('../setup');
+
+const {_toDependencies} = require('../../app/plugins');
+
+describe('plugins', function () {
+  describe('#toDependencies()', function () {
+    it('should convert dependencies form hyperterm\'s format to npm\'s', function () {
+      const plugins = ['project1', 'project2#1.0.0', 'project3@beta',
+                       '@org1/project4#1.0.0', '@org2/project5@alpha',
+                       '@org3/project6'];
+
+      const result = _toDependencies({plugins: plugins});
+      result.should.have.property('project1', 'latest');
+      result.should.have.property('project2', '1.0.0');
+      result.should.have.property('project3', 'beta');
+      result.should.have.property('@org1/project4', '1.0.0');
+      result.should.have.property('@org2/project5', 'alpha');
+      result.should.have.property('@org3/project6', 'latest');
```

Comment:
I know that this is kind of an annoying drive-by-suggestion, so sorry in advance; this is just such a good example of why I like the unexpected assertion framework.

The above assertions could be folded into a single one:

```js
expect(_toDependencies({ plugins: plugins }), 'to satisfy', {
  'project1': 'latest',
  'project2': '1.0.0',
  'project3': 'beta',
  '@org1/project4': '1.0.0',
  '@org2/project5': 'alpha',
  '@org3/project6': 'latest'
});
```

The `to satisfy` assertion is like equals - except that it doesn't complain about extra keys. So the following won't fail:

```js
expect({ foo: 'bar', qux: 'baz' }, 'to satisfy', { foo: 'bar' });
```

Rolling multiple assertions, like in this example with should, into a single one, means that the error can give you a lot more context and generally a better idea about what is wrong. The focus of the unexpected assertion library is to provide good enough tools that you can fold these assertions together and to give the best possible output when the assertions fail.

```js
> expect({ foo: 'bar', baz: 'qux' }, 'to satisfy', { foo: 'qux' });

expected { foo: 'bar', baz: 'qux' } to satisfy { foo: 'qux' }

{
  foo: 'bar', // should equal 'qux'
              //
              // bar
              // qux
  baz: 'qux'
}
```

http://unexpected.js.org/

(I work/worked with the authors)